runtime/rasdump/javadump.cpp

```diff
@@ -1770,12 +1770,55 @@ JavaCoreDumpWriter::writeMemoryCountersSection(void)
 			writeNativeAllocator("Other", i + 1, 0, frame->liveBytes, frame->liveAllocations);
 		}
 	}
+        /* Write the section trailer */
+        _OutputStream.writeCharacters(
+                "NULL\n"
+                "NULL           ------------------------------------------------------------------------\n"
+        );
+
+#ifdef J9ZTPF
+	if (mmap(NULL, 0, 0, MAP_SUPPORTED, 0, 0) != MAP_FAILED)  {
+		rc = mprivatestats(&ztpfNativeStats);
+
+		if (rc == 0)  {
+			_OutputStream.writeCharacters("0SECTION		Native configuration information for z/TPF\n");
+			_OutputStream.writeCharacters("NULL		==========================================\n");
+			_OutputStream.writeCharacters("1TPFHWMAXX	Highwater mark 64-bit heap 1MB Frames (limited by MAXXMMES): ");
+			_OutputStream.writeInteger(ztpfNativeStats.totalMAXXMMESRegion1MBFrames, "%u\n");
+			_OutputStream.writeCharacters("1TPFLIMMAXX	MAXXMMES keypoint A setting: ");
+			_OutputStream.writeInteger(ecbp2()->ce2proc->iproc_maxheap_64, "%u\n");
+			_OutputStream.writeCharacters("1TPFHWMMAP 	Highwater mark 64-bit MMAP heap 1MB Frames (limited by MAXMMAP): ");
+			_OutputStream.writeInteger(ztpfNativeStats.totalMMAPRegion1MBFrames, "%u\n");
+			_OutputStream.writeCharacters("1TPFLIMMMAP	MAXMMAP process setting (either keypoint A MAXMMAP or com.ibm.tpf.maxmmap property):       ");
+			_OutputStream.writeInteger(ztpfNativeStats.currentProcessMAXMMAPLimit, "%u\n");
+			_OutputStream.writeCharacters("1TPFTOTHW64	Total highwater mark 64-bit heap 1MB Frames (limited by MAXXMMES+MAXMMAP): ");
+			_OutputStream.writeInteger(ztpfNativeStats.total64bitHeap1MBFrames, "%u\n");
+			_OutputStream.writeCharacters("1TPFHWEMPS	Total highwater mark 31-bit heap 1MB Frames (limited by EMPS): ");
+			_OutputStream.writeInteger(ztpfNativeStats.total31bitHeap1MBFrames, "%u\n");
+			_OutputStream.writeCharacters("1TPFLIMEMPS	EMPS keypoint A setting: ");
+			_OutputStream.writeInteger(ecbp2()->ce2proc->iproc_maxheap_31, "%u\n");
+			_OutputStream.writeCharacters("1TPFHWGC		Total highwater mark GC heap 1MB Frames (limited by -Xmx): ");
+			_OutputStream.writeInteger(ztpfNativeStats.totalGCHeap1MBFrames, "%u\n");
+			_OutputStream.writeCharacters("1TPFHWJITCC	Total highwater mark JIT Code Cache heap 1MB Frames: ");
+			_OutputStream.writeInteger(ztpfNativeStats.totalJITCodeCache1MBFrames, "%u\n");
+			_OutputStream.writeCharacters("1TPFMMSTART	MMAP Region Virtual Address start: ");
+			_OutputStream.writePointer(ztpfNativeStats.mmapRegionVirtualAddressStart);
+			_OutputStream.writeCharacters("\n");
+			_OutputStream.writeCharacters("1TPFMMEND	MMAP Region highwater mark Virtual Address (limited by CINFC_CMMEVEN): ");
+			_OutputStream.writePointer(ztpfNativeStats.mmapRegionVirtualAddressEnd);
+			_OutputStream.writeCharacters("\n");
+			_OutputStream.writeCharacters("1TPFLIMMTHD	MTHD Keypoint A setting: ");
+			_OutputStream.writeInteger(*((short *)(cinfc_fast(CINFC_CMMTHMAX))), "%u\n");
```

Comment:
This has a redundant pair of parentheses (to the right of the dereference), casts are missing to arrive at `UDATA` expected by `writeInteger()` and the `%u` format is inappropriate for a `short` argument:
```c
			_OutputStream.writeInteger((UDATA)(IDATA)*(short *)(cinfc_fast(CINFC_CMMTHMAX)), "%zd\n");
```